clients/secret/secret.go

```diff
@@ -0,0 +1,165 @@
+package secret
+
+import (
+	"context"
+	"fmt"
+	"github.com/influxdata/influx-cli/v2/api"
+	"github.com/influxdata/influx-cli/v2/clients"
+)
+
+type Client struct {
+	clients.CLI
+	api.SecretsApi
+	api.OrganizationsApi
+}
+
+type secretPrintOpt struct {
+	deleted bool
+	secret  secret
+	secrets []secret
+}
+
+type secret struct {
+	key   string
+	orgID string
+}
+
+type DeleteParams struct {
+	clients.OrgParams
+	Key string
+}
+
+func (c Client) Delete(ctx context.Context, params *DeleteParams) error {
+	orgID, err := c.getOrgID(ctx, params.OrgParams)
+	if err != nil {
+		return err
+	}
+
+	if err := c.PostOrgsIDSecrets(ctx, orgID).Execute(); err != nil {
+		return fmt.Errorf("failed to delete secret with key %q: %v", params.Key, err)
+	}
+
+	return c.printSecrets(secretPrintOpt{
+		deleted: true,
+		secret: secret{
+			key:   params.Key,
+			orgID: orgID,
+		},
+	})
+}
+
+type ListParams struct {
+	clients.OrgParams
+}
+
+func (c Client) List(ctx context.Context, params *ListParams) error {
+
+	orgID, err := c.getOrgID(ctx, params.OrgParams)
+	if err != nil {
+		return err
+	}
+
+	response, err := c.GetOrgsIDSecrets(ctx, orgID).Execute()
+	if err != nil {
+		return fmt.Errorf("failed to retrieve secret keys: %s", err)
+	}
+
+	secrets := response.GetSecrets()
+	opts := make([]secret, 0, len(secrets))
+	for _, entry := range secrets {
+		opts = append(opts, secret{
+			key:   entry,
+			orgID: orgID,
+		})
+	}
```

Comment:
Since you know the exact length, you could instead:
```suggestion
	opts := make([]secret, len(secrets))
	for i, entry := range secrets {
		opts[i] = secret{
			key:   entry,
			orgID: orgID,
		}
	}
```